### PR TITLE
Add Discord notification

### PR DESCRIPTION
**Pull request by ianepperson**
Adds discord to the notification options using [Discord's webhooks](https://discord.com/developers/docs/resources/webhook).

Environment variables:

- Required:
  DISCORD_WEBHOOK_URL
- Optional:
 DISCORD_USERNAME - if empty, defaults to the defined username
 DISCORD_AVATAR_URL - if empty, defaults to the defined avatar

- [ ] TODO after approval: update the Wiki notify page: https://github.com/acmesh-official/acme.sh/wiki/notify

### CODE REVIEW

**Comment by ianepperson** [comment]
Almost entirely copied and modified from the Slack notification code.

File: notify/discord.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    _data="$_data, \"username\": \"$DISCORD_USERNAME\" "
+  fi
+  if [ -n "$DISCORD_AVATAR_URL" ]; then
+    _data="$_data, \"avatar_url\": \"$DISCORD_AVATAR_URL\" "
+  fi
+  _data="$_data}"
+
+  if _post "$_data" "$DISCORD_WEBHOOK_URL?wait=true"; then
+    # shellcheck disable=SC2154
+    if [ -n "$response" ]; then
```
Comment (ianepperson) [inline notify/discord.sh:49]: Discord returns an HTTP error if there's a problem and never seems to return any content if there isn't. Success responses contain the JSON message of what was posted.

**Comment by darthShadow** [comment]
Is there any issue with this PR? If not, can we get this merged soon?

**Comment by darthShadow** [comment]
Sorry for the delay, updated the wiki: https://github.com/acmesh-official/acme.sh/wiki/notify#19-set-notification-for-discord-webhooks

**Comment by Neilpang** [comment]
what is DISCORD_CHANNEL  in the wiki page?

**Comment by darthShadow** [comment]
Oops ... 😅 

Should be fixed now.

**Pull request merged**